typescript/schedules.md

```diff
@@ -0,0 +1,1257 @@
+   */
+  jitter?: Milliseconds;
+
+  /**
+   * IANA timezone name, for example `US/Pacific`.
+   * 
+   * https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
+   * 
+   * @default UTC
+   */
+  timezone?: string;
+
+  // Add to SDK if requested by users:
+  //
+  // bytes timezone_data = 11;
+  //
+  // Time zone to interpret all CalendarSpecs in.
+  //
+  // If unset, defaults to UTC. We recommend using UTC for your application if
+  // at all possible, to avoid various surprising properties of time zones.
+  //
+  // Time zones may be provided by name, corresponding to names in the IANA
+  // time zone database (see https://www.iana.org/time-zones). The definition
+  // will be loaded by the Temporal server from the environment it runs in.
+  //
+  // If your application requires more control over the time zone definition
+  // used, it may pass in a complete definition in the form of a TZif file
+  // from the time zone database. If present, this will be used instead of
+  // loading anything from the environment. You are then responsible for
+  // updating timezone_data when the definition changes.
+  //
+  // Calendar spec matching is based on literal matching of the clock time
+  // with no special handling of DST: if you write a calendar spec that fires
+  // at 2:30am and specify a time zone that follows DST, that action will not
+  // be triggered on the day that has no 2:30am. Similarly, an action that
+  // fires at 1:30am will be triggered twice on the day that has two 1:30s.
+}
+
+export type StartWorkflowAction<Action extends Workflow> = Omit<
+  WorkflowStartOptions<Action>,
+  'workflowIdReusePolicy' | 'cronSchedule' | 'followRuns'
+> & {
+  /**
+   * Metadata that's propagated between workflows and activities? TODO someone explain headers to me so I can improve
+   * this apidoc and maybe:
+   * - remove from here and only expose to interceptor?
+   * - or make it Record<string, unknown> and apply client's data converter?
+   */
+  headers: Headers;
+};
+
+export type ScheduleActionType = Workflow;
+
+/**
+ * Currently, Temporal Server only supports {@link StartWorkflowAction}.
+ */
+export type ScheduleActionOptions<Action extends ScheduleActionType = ScheduleActionType> = StartWorkflowAction<Action>;
+// in future:
+// type SomethingElse = 'hi'
+// type ScheduleAction = Workflow | SomethingElse
+// type ExpectsSomethingElse<Action extends SomethingElse> = Action extends SomethingElse ? Action : number;
+// type StartSomethingElseAction<Action extends SomethingElse> = ExpectsSomethingElse<Action>
+// type ScheduleActionOptions<Action extends ScheduleAction> = Action extends Workflow
+//   ? StartWorkflowAction<Action>
+//   : Action extends SomethingElse
+//     ? StartSomethingElseAction<Action>
+//     : never;
+
+export type HandleFor<T> = T extends Workflow ? WorkflowHandle<T> : never;
+
+/**
+ * Options for starting a Workflow
+ */
+export interface ScheduleOptions<Action extends ScheduleActionType> {
```

Comment:
Can we make this as close to the describe response as possible?

Reply:
Why? Copying? 

Differences are: 

- `ScheduleDescription` doesn't have unions, like this:

<img width="1561" alt="image" src="https://user-images.githubusercontent.com/251288/184466176-74bffef7-b252-4c87-a261-33805f6a5bd0.png">

- `ScheduleDescription` has all the ScheduleInfo fields
- `ScheduleDescription` has `actionsAreLimited`
- `ScheduleOptions` has `triggerImmediately` and `backfill`